daffodil-udf/src/main/java/org/apache/daffodil/udf/UDFunctionProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.udf;
+
+public abstract class UDFunctionProvider {
```

Comment:
java doc please. Is there going to be more than one instance of this class? (Maybe for testing reasons?) otherwise I'd suggest eliminating this abstract class and just having a single concrete one. 

Naming again. Suggest UDFProvider or UserDefinedFunctionProvider.

Reply:
I will add the javadoc. And yes, each UDF Provider that we poll during compilation will be an instance of this class.